File: src/test/java/org/skefir/data/EventHistoryColumn.java
package org.skefir.data;

import lombok.Getter;

public enum EventHistoryColumn implements DataTableColumn {
    DATE("Date (GMT)"),
    REFERENCE("Reference"),
    ACTUAL("Actual"),
    FORECAST("Forecast"),
    PREVIOUS("Previous");

    @Getter
    private final String title;

    EventHistoryColumn(String title) {
        this.title = title;
    }
}

File: src/test/kotlin/org/skefir/data/EventHistoryColumn.kt
package org.skefir.data

enum class EventHistoryColumn(private val title: String) : DataTableColumn {
    DATE("Date (GMT)"),
    REFERENCE("Reference"),
    ACTUAL("Actual"),
    FORECAST("Forecast"),
    PREVIOUS("Previous");

    override fun getTitle(): String = title

    override fun getColumnByTitle(title: String): DataTableColumn =
        EventHistoryColumn.values().first { it.title == title }

}